### PR TITLE
Sync with 2239 and 2235

### PR DESCRIPTION
**Pull request by shargon**
- Require https://github.com/neo-project/neo/pull/2239
- Require https://github.com/neo-project/neo/pull/2235

### CODE REVIEW

**Comment by erikzhang** [comment]
And the templates.

**Comment by shargon** [comment]
Wait for the last nuget with the nef changes

File: templates/Template.CSharp/Contract1.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public static byte[] MyMethod()
             return Storage.Get("Hello");
         }
 
-        public static void _deploy(bool update)
+        public static void _deploy(bool update, object data)
```
Comment (erikzhang) [inline templates/Template.CSharp/Contract1.cs:30]:
```suggestion
        public static void _deploy(object data, bool update)
```

**Comment by Tommo-L** [comment]
UT failed

**Comment by shargon** [comment]
Ready

**Review by erikzhang — APPROVED** [review]

**Pull request merged**